DOCS 6027 - Added Naming Rules for Variables section. Added list of reserved words. (#873)
mulesoft/docs-mule-runtime-3: modules/ROOT/pages/dataweave-variables.adoc

You can combine several local variable definitions as a comma separated list inside the using function. For example: `using (firstName='Annie', lastName='Point')`

== Naming Rules for Variables

> You can declare any variable name as long as it is a valid identifier. See xref:dataweave-language-introduction.adoc#valid-identifiers[Rules for Declaring Valid Identifiers] for more details.

Reviewer: ```suggestion
You can declare any variable name that is a valid identifier. See xref:dataweave-language-introduction.adoc#valid-identifiers[Rules for Declaring Valid Identifiers] for more details.
```

You can combine several local variable definitions as a comma separated list inside the using function. For example: `using (firstName='Annie', lastName='Point')`

== Naming Rules for Variables

> Variable names have certain requirements to be valid, following are the rules for naming variables:

Reviewer 2: grammar issue with comma there. Suggest something like "Variable names must meet the following requirements to be valid:" or something similar.

* All variable names must begin with a letter of the alphabet (a-Z).
* After the first letter, variable names can contain any combination of letters, numbers and underscores (`_`).
* The variable name cannot be any DataWeave reserved keyword (see <<reserved-keywords>> for a complete list).

Below are some examples of valid variable names:

* `myType`
* `abc123`
* `a1_3BC_22`
* `Z_______4`
* `F`

> The following table has some examples of invalid variable names, and a description of what makes them invalid:

Reviewer 2: suggest "has some examples" -> "provides examples" 

"names, and" -> "names and"

[%header%autowidth.spread,cols="a,a"]
|===
| Variable name | Issue
| `123456` | Cannot start with a number.
| `value$` | Cannot contain special characters.
| `_number` | Cannot start with an underscore.
| `type` | Cannot be a reserved keyword.
|===

[[reserved-keywords]]
=== Reserved Keywords

The following list contain all reserved keywords that cannot be used as variable names:

> * if

Reviewer 2: all should probably be in code font, surrounded by backticks. Great to have this list!!

* else
* unless
* using
* as
* is
* null
* true
* false
* default
* case
* fun
* input
* output
* ns
* type
* import
* var
* and
* or

> * throw

Reviewer 2: is it "throw" or "throws"? Prob similar to "throw" keyword in Scala, I see.

Author: throw is ok, I got this list from Mariano, picked directly from the validation code.
